### ISSUE TITLE
installation fails on Windows 10

### ISSUE THREAD

**Issue by afunix** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#3688]
**Checklist:**

* [x] I have checked that there are no issues with similar or the same content

  **YOU SHOULD CHECK CLOSED ISSUES ASWELL**
* [x] I have checked the FAQ (https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/wiki/FAQ) and the answer I am looking for is not there
* [x] I have double checked and can reproduce the issue

**OS:** Windows 10 x64 1903 (18362.476)

**GPMDP Version:** 4.6.1

**Issue Descriptions:**
Installation fails on Windows 10.
Player executable fails to start.

**Steps to Reproduce:**
1. Start setup executable downloaded from official website
2. After some time an error window appears with an option to view logs
A number of System.NullReferenceException, System.IO.DirectoryNotFoundException and System.ComponentModel.Win32Exception are in the log.
[SquirrelSetup.log](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/files/3892867/SquirrelSetup.log)

**Comment by z3rods** [comment]
I, too, am having an issue installing the program. I recently had issues attempting to access Play Music itself (the app would get stuck on the library loading and never progress any further; the YouTube Music part still worked). 

So I uninstalled the app through the Programs and Features applet and reinstalled. Had the same issue. so I decided to delete the app's folder from my user data folders, and now I get this error. 
![image](https://user-images.githubusercontent.com/15926276/69931893-eb16a080-1496-11ea-9176-dc71dcd15b67.png)

**Comment by eliotcole-projectfive** [comment]
I'm getting the same issue that @z3rods mentions in his second sentence, but I don't use any of the paid services, I just want to access my files that are on Google. I have reinstalled (gr), and restarted both the application, and Windows, fully.

When the program starts it just stops at the "Loading Music Library…" screen.

Via the system tray I've tried it with both "_JSON API_" (default 'on' it seems) and "_Playback API_" in varying states of running, after full restarts. Nothing.

Could it be something to do with the "NET::ERR_CERT_DATE_INVALID" on the **https googleplaymusicdesktopplayer dot com** site? Assuming this is a Chrome App, perhaps Chrome is stopping the application make a handshake or something with the host site?

I'll ping Samuel on Twitter about the site, at any rate, I'm sure he'd want to know.

Best
E
> I, too, am having an issue installing the program. I recently had issues attempting to access Play Music itself (the app would get stuck on the library loading and never progress any further; the YouTube Music part still worked).
> 
> So I uninstalled the app through the Programs and Features applet and reinstalled. Had the same issue. so I decided to delete the app's folder from my user data folders, and now I get this error.
> ![image](https://user-images.githubusercontent.com/15926276/69931893-eb16a080-1496-11ea-9176-dc71dcd15b67.png)
> 
> Steps to reproduce:
> 
> 1. Have error loading music library (no idea how to produce this).
> 2. Delete all relevant app data.
> 3. Reinstall the program.
> 
> I am running Windows 10 Pro version 1903, build 18362.476.

**Comment by jostrander** [comment]
I don't have the ability to test a fresh install at the moment on Windows. Can you uninstall the application through the control panel and remove `GPMDP_3` from `%appdata%` and `Google Play Music Desktop Player` from `%localappdata%` (or vis versa) and see if you can reinstall then?

**Comment by z3rods** [comment]
If you're asking me, I've already done that. That's when I get this error.

**Comment by jostrander** [comment]
The logs from the original issue creator seem to indicate some kind of access denied error during installation:

```
2019-11-25 09:49:19> CheckForUpdateImpl: Reading RELEASES file from C:\Users\cqx874\AppData\Local\SquirrelTemp
2019-11-25 09:49:19> CheckForUpdateImpl: First run or local directory is corrupt, starting from scratch
2019-11-25 09:49:22> ApplyReleasesImpl: Writing files to app directory: C:\Users\cqx874\AppData\Local\GPMDP_3\app-4.6.1
2019-11-25 09:49:28> ApplyReleasesImpl: Squirrel Enabled Apps: [C:\Users\cqx874\AppData\Local\GPMDP_3\app-4.6.1\Google Play Music Desktop Player.exe]
2019-11-25 09:49:31> ApplyReleasesImpl: Couldn't run Squirrel hook, continuing: C:\Users\cqx874\AppData\Local\GPMDP_3\app-4.6.1\Google Play Music Desktop Player.exe: System.ComponentModel.Win32Exception (0x80004005): Access is denied
```
Is it possible file locker (I think this is what Windows 10 calls it?) is preventing GPMDP from writing to the install directories? It's part of the windows 10 protection suite that's built in. I've had issues with other programs and this particular feature.

**Comment by z3rods** [comment]
I hadn't thought of that (since no issue had popped up before), so I disabled WFP and tried to install again (using 4.6.1). Upon attempting to sign in, I got this error from Google. 
![image](https://user-images.githubusercontent.com/15926276/70203193-1b09b200-16ea-11ea-84b8-84525d3007f5.png)

**Comment by jostrander** [comment]
Yes. That's a known issue that popped up somewhat recently. See issue #3545. Otherwise, that solved your issue?

**Comment by z3rods** [comment]
Sure, but it feels off to me that one has to disable a system protection feature that many may not even know about to get it to install. It's made even more disconcerting that the only way I could find to disable involves digging into the registry. 

Oh well. I guess I'll have to install one of those other builds in order to get around the new error I'm having.

**Comment by jostrander** [comment]
Hmm. I don't disagree with your sentiment. Can you elaborate on what you had to do to get it installed so I or someone else reading can recreate or debug the issue?

**Comment by z3rods** [comment]
Sure thing. 

Make sure you've uninstalled the program via Control Panel/Settings app. Also delete the folders 
`GPMDP_3` from `C:\Users\<username>\AppData\Local` and `Google Play Music Desktop Player` from `C:\Users\<username>\AppData\Roaming` (that's where these folders were located for me).

To disable Windows File Protection, follow these steps:

1. Open Regedit.
2. Navigate to **HKEY_LOCAL_MACHINE** > **SOFTWARE** > **Microsoft** > **Windows NT** > **CurrentVersion** > **Winlogon**.
3. Look for the key **"SFCDisable"** (or create it by clicking on the **Winlogon** folder and choosing **"New** > **DWORD (32-bit value)**") and set it to **FFFFFF9D**. This disables it completely and was the only value that worked consistently for me (setting it to 1 or 2 disables it, but apparently only when you have a physical kernel debugger attached according to my research and subsequent experience; YRMV).
4. Reboot Windows.

Upon booting back up, you should be able to install the program with no problem. However, I still run into the error I attached above, even with the newer builds from AppVeyor.

**Comment by jostrander** [comment]
Thanks for the write up. Just did a quick Google since I don't have access to win 10 right now.
Did GPMDP show up in the [Controlled Folder Access](https://support.microsoft.com/en-us/help/4046851/windows-10-allow-blocked-app-windows-security) part of Windows Defender? 
This is where I thought it may have been getting caught since the app data directories are starting to become locked down. 

This says it needs to be enabled but I'm pretty sure it's getting enabled by default on some installs.
https://docs.microsoft.com/en-us/windows/security/threat-protection/microsoft-defender-atp/enable-controlled-folders

**Comment by eliotcole-projectfive** [comment]
Ola, just catching up on the stuff you guys are doing, sorry for being a bit dumb, but this all seems like stuff that I've no idea about.

Is this resolving the installation issues, or the library loading issues?

Is this part referring to the library loading issues?
> Yes. That's a known issue that popped up somewhat recently. See issue #3545. Otherwise, that solved your issue?

**Comment by jostrander** [comment]
@eliotcole-projectfive still figuring out installation issues, and that is referring to sign in being blocked. For library loading see #3660. That part has been fixed but not versioned and released.

**Comment by eliotcole-projectfive** [comment]
> @eliotcole-projectfive still figuring out installation issues, and that is referring to sign in being blocked. For library loading see #3660. That part has been fixed but not versioned and released.

_Sweeeeeeeeeeeet_ as a nut. Cheers, @jostrander ... you are the don.

Sorry for bothering (and my brain thanks you for finding the issue that I couldn't find)!

**Comment by z3rods** [comment]
> 
> 
> Thanks for the write up. Just did a quick Google since I don't have access to win 10 right now.
> Did GPMDP show up in the [Controlled Folder Access](https://support.microsoft.com/en-us/help/4046851/windows-10-allow-blocked-app-windows-security) part of Windows Defender?
> This is where I thought it may have been getting caught since the app data directories are starting to become locked down.
> 
> This says it needs to be enabled but I'm pretty sure it's getting enabled by default on some installs.
> https://docs.microsoft.com/en-us/windows/security/threat-protection/microsoft-defender-atp/enable-controlled-folders

I don't even have the option to toggle Controlled Folder Access. Windows Defender is disabled on my system due to the fact that I roll my own anti-virus and anti-malware programs (Webroot and Malwarebytes). I suppose I could try to disable those, but I doubt they're having any effect since they don't throw any warnings or dialogues of any kind when attempting to install the program.

**Comment by timhobbs** [comment]
You may not have to go the regedit route. I was having the same issue and when I saw the mention about  file protection that was where I tried looking. All you have to do is right-click and "Unblock" the file. Once I did this I was able to install, but I still run into a "Loading.." message that just sort of hangs.

**Comment by roryhewitt** [comment]
@timhobbs Sorry - "right-click" _which_ file and unblock? You mean the installer .exe?

**Comment by timhobbs** [comment]
@roryhewitt  - correct. you can do this for any file that you download that is "blocked" by windows. just select properties and then down at the bottom you'll see a message and an unblock button, like in this example:
<img src="https://thewindowsclub-thewindowsclubco.netdna-ssl.com/wp-content/uploads/2012/08/unblock.jpg" width=300/>

**Comment by jostrander** [comment]
Is this still happening with 4.7.1?

**Comment by roryhewitt** [comment]
Seems to be resolved now.
